docs/getting-started-guides/docker-multinode.md

```diff
@@ -74,54 +78,165 @@ This pattern is necessary because the `flannel` daemon is responsible for settin
+export NODES="vcap@10.10.102.153"
+export MASTER="vcap@10.10.102.150"
+```
 
-See [here](docker-multinode/master.md) for detailed instructions explanation.
+### Support different infrastructures
 
-## Adding a worker node
+We use `baremetal` as default infrastructure provider, which means we only require a batch of machines (VMs or physical servers) with Docker installed, and then deploy k8s on them remotely by using `scp` & `ssh`.
 
-Once your master is up and running you can add one or more workers on different machines.
+But we also planned to supported any other infrastructure like GCE, AWS etc. So there will be more provider specific versions based on the existing [docker-baremetal](../../cluster/docker-baremetal/) implementation, e.g. `docker-gce` or `docker-aws`, to support `gcloud ssh` or AWS identity.
 
-Clone the Kubernetes repo, and run [worker.sh](docker-multinode/worker.sh) on the worker machine with root:
+And then tell scripts you want to deploy on other infrastructure like this:
 
 ```sh
-export MASTER_IP=<your_master_ip (e.g. 1.2.3.4)>
-cd kubernetes/docs/getting-started-guides/docker-multinode/
-./worker.sh
+export NODES="vcap@10.10.102.152"
+export MASTER="vcap@10.10.102.150"
+export KUBERNETES_PROVIDER=docker
+export INFRA=gce
+./kube-up.sh
+```
+
+
+Only a few functions need to be rewrite, so welcome to contribute!
+
+## Test it out
+
+On every node, you can see there are two containers running by `docker ps`:
+
+```console
+kube_in_docker_proxy_xxx
+kube_in_docker_kubelet_xxx
 ```
 
-`Worker done!`
+And on Master node, you can see extra master containers running:
+
+```console
+k8s_scheduler.xxx
+k8s_apiserver.xxx
+k8s_controller-manager.xxx
+```
 
-See [here](docker-multinode/worker.md) for detailed instructions explanation.
+As we use `hyperkube` image to run k8s, we **do not** need to compile binaries, please download and extract `kubectl` binary from [releases page](https://github.com/kubernetes/kubernetes/releases).
 
-## Deploy a DNS
+At last, use `$ kubectl get nodes` to see if all of your nodes are ready.
+
+```console
+$ kubectl get nodes
+NAME            LABELS                                 STATUS
+10.10.102.150   kubernetes.io/hostname=10.10.103.150   Ready
+10.10.102.153   kubernetes.io/hostname=10.10.102.153   Ready
+```
+
+### Deploy a DNS
 
 See [here](docker-multinode/deployDNS.md) for instructions.
 
-## Testing your cluster
+Then you can run Kubernetes [guest-example](../../examples/guestbook/) to build a Redis backend cluster on the k8s．
+
+## Customize
+
+One of the biggest benefits of using Docker to run Kubernetes is users can customize the cluster freely before you running `kube-up.sh`:
+
+### master
+
+You need to enable **customized master mode** before deploy:
+
+```sh
+export MASTER_CONF=yes
+```
+
+The configuration file of Master locates in `docker/kube-config/master-multi.json`, which will be mounted as volume for Master Pod to consume, you can modify it freely **before deploying**.
```

Comment:
i think this should be `cluster/images/hyperkube/master-multi.json` according to https://github.com/ZJU-SEL/kubernetes/blob/1219e37/cluster/docker-baremetal/util.sh#L74 ?